### PR TITLE
jlamillan/support oel 7.4

### PR DESCRIPTION
**Pull request by jlamillan**
This change _converts_ the shell scripts from supporting Ubuntu to OEL 7 that Joe did. I cleaned up some rough edges, updated OEL to 7.4, and did some testing.

Longer term, the goal is to move away from shell scripts to some configuration management tool, which will hopefully make it easier for us to support multiple OS's (without adding an egregious amount of brittle if / else statements all over the place).

I ran the Kubernetes conformance tests (123 Passed | 0 Failed) against this new cluster config in two separate regions without issue. I've also did some additional manual testing around pod networking since I made some additional changes in that area to reduce some duplicate code.

Once this is merged, we should probably create a new tag so that v.1.0.0 can be the "initial" release that supported Ubuntu and this would be a subsequent release that supports OEL.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -164,14 +164,14 @@ worker_ssh_ingress = "0.0.0.0/0"
 $ terraform output ssh_private_key > generated/instances_id_rsa
 # Retrieve public IP for etcd nodes
 $ terraform output etcd_public_ips
-# Log in as user ubuntu to the Canonical Ubuntu OS
-$ ssh -i `pwd`/generated/instances_id_rsa ubuntu@ETCD_INSTANCE_IP
+# Log in as user opc to the OEL OS
+$ ssh -i `pwd`/generated/instances_id_rsa oel@ETCD_INSTANCE_IP
```
Comment (jrosinsk) [inline README.md]: user looks weird here  'oel'    should be 'opc'  right.

File: instances/etcd/cloud_init/bootstrap.template.sh [hunk 2]
```diff
@@ -62,7 +69,3 @@ done
 
 # put the flannel config in etcd
 curl -sf -L http://$FQDN_HOSTNAME:2379/v2/keys/flannel/network/config -X PUT --data-urlencode value@/tmp/flannel-network.json
```
Comment (jrosinsk) [inline instances/etcd/cloud_init/bootstrap.template.sh]: Did you verify that the home dir permissions are correct before removing this?  ls -la /home/  to make sure opc isn't owned by root

File: instances/k8smaster/main.tf [hunk 3]
```diff
@@ -12,12 +12,14 @@ resource "oci_core_instance" "TFInstanceK8sMaster" {
   shape               = "${var.shape}"
   subnet_id           = "${var.subnet_id}"
 
-  metadata {
+  extended_metadata {
     roles               = "masters"
     ssh_authorized_keys = "${var.ssh_public_key_openssh}"
     user_data           = "${data.template_cloudinit_config.master.rendered}"
+    tags = "group:k8s-master"
```
Comment (jrosinsk) [inline instances/k8smaster/main.tf]: this looks like an ansible tag.

File: instances/k8sworker/main.tf [hunk 4]
```diff
@@ -12,10 +12,11 @@ resource "oci_core_instance" "TFInstanceK8sWorker" {
   shape               = "${var.shape}"
   subnet_id           = "${var.subnet_id}"
 
-  metadata {
+  extended_metadata {
     roles               = "nodes"
     ssh_authorized_keys = "${var.ssh_public_key_openssh}"
     user_data           = "${data.template_cloudinit_config.master.rendered}"
+    tags = "group:k8s-worker"
```
Comment (jrosinsk) [inline instances/k8sworker/main.tf]: looks like an ansible tag.  is this used.

File: instances/k8sworker/scripts/setup.template.sh [hunk 5]
```diff
@@ -4,13 +4,10 @@ EXTERNAL_IP=$(curl -s -m 10 http://whatismyip.akamai.com/)
 NAMESPACE=$(echo -n "${domain_name}" | sed "s/\.oraclevcn\.com//g")
 FQDN_HOSTNAME=$(getent hosts $(ip route get 1 | awk '{print $NF;exit}') | awk '{print $2}')
 
-# make sure ubuntu owns home dir
-chown ubuntu:ubuntu /home/ubuntu
-
-# pull instance metadata
+# Pull instance metadata
```
Comment (jrosinsk) [inline instances/k8sworker/scripts/setup.template.sh]: make sure proper ownership of opc home dir

File: instances/etcd/cloud_init/bootstrap.template.sh (hunk 2 above)
Comment (jlamillan) [reply instances/etcd/cloud_init/bootstrap.template.sh]: Yes I did. I think Ubuntu had some sort of bug.

File: instances/k8smaster/main.tf (hunk 3 above)
Comment (jlamillan) [reply instances/k8smaster/main.tf]: I purposely added these tags to illustrate using the new extended_metadata fields. I could remove the `group` portion to make it more generic.

File: instances/k8sworker/scripts/setup.template.sh (hunk 5 above)
Comment (jlamillan) [reply instances/k8sworker/scripts/setup.template.sh]: It's fine.

```
$ ls -all /home/
total 8
drwxr-xr-x.  3 root root   17 Oct 13 20:41 .
dr-xr-xr-x. 17 root root 4096 Sep 26 18:34 ..
drwx------.  5 opc  opc  4096 Oct 13 20:50 opc
```

File: README.md (hunk 1 above)
Comment (jlamillan) [reply README.md]: Good eye. Pushed [fix](https://github.com/oracle/terraform-kubernetes-installer/commit/ca08a74c80a0f1d588a21ce7fe8fc5391ea9519e).

File: instances/etcd/cloud_init/bootstrap.template.sh (hunk 2 above)
Comment (jrosinsk) [reply instances/etcd/cloud_init/bootstrap.template.sh]: sounds good.

File: instances/k8smaster/main.tf (hunk 3 above)
Comment (jrosinsk) [reply instances/k8smaster/main.tf]: ok.  looks good

**Comment by jrosinsk** [comment]
looks good

**Comment by jlamillan** [comment]
I just found another issue: during teardown, our scripts are depending on a command called `ip`, which is not there for OEL.
